cmake/ssl.cmake

```diff
@@ -33,14 +33,11 @@ if(gRPC_SSL_PROVIDER STREQUAL "module")
     if(TARGET ssl)
       set(_gRPC_SSL_LIBRARIES ssl)
       set(_gRPC_SSL_INCLUDE_DIR ${BORINGSSL_ROOT_DIR}/include)
+      install(TARGETS ssl crypto EXPORT gRPCTargets)
```

Comment:
I remember we used to get an error that we are trying to install a target coming from a different project if we tried to install grpc when the dependencies were added via add_subdirectory.  I assume the reason for this line to overcome that problem?

Reply:
Here's the error that occurs (at configure time) if you don't include ssl in the gRPCTargets export:

```
CMake Error:
install(EXPORT "gRPCTargets" ...) includes target "grpc"
which requires target "ssl" that is not in any export set.

```

Reply:
Yes, that's the main reason why we forbid the gRPC_INSTALL with provider=submodule in the past. I didn't know that it's that easy to work around.
Are there any important consequences of doing so?